codecov_auth/authentication/repo_auth.py

```diff
@@ -43,10 +43,15 @@ def repo_auth_custom_exception_handler(exc, context):
     give the user something better than "Invalid Token" or "Authentication credentials were not provided."
     """
     response = exception_handler(exc, context)
-    if response is not None:
+    if (
+        response is not None
+        and hasattr(response, "status_code")
+        and response.status_code == 401
+        and hasattr(response, "data")
+    ):
```

Comment:
these `hasattr`s are making me think we're sometimes getting something that is a totally different type than we're expecting. i am confused

the `KeyError`s in sentry were just about that `detail` key, were you also seeing errors about `status_code` and `data` not even existing? the link in the PR description is just to the project, not a specific issue

Reply:
you right -  I think lots of types of exceptions are flowing through here but this is an overreaction, to make it fully bomb-proof, so that this type of issue can't arise again. It's doing the safe checks for every element of the object that we check - just because I was thinking, what if I just patch this KeyError with `detail` and another similar issue arises with this block.

What do you think - should I pull it back to just the KeyError, or should I leave this super cautious logic?

Reply:
Added some in-line comments to reflect my thoughts here, but going to merge so that the issue can be fixed.